pkg/cmd/channel/channel.go

```diff
@@ -0,0 +1,230 @@
+package channel
+
+import (
+	"encoding/json"
+	"errors"
+	"text/tabwriter"
+	"time"
+
+	stream "github.com/GetStream/stream-chat-go/v5"
+	"github.com/GetStream/stream-cli/pkg/config"
+	"github.com/cheynewallace/tabby"
+	"github.com/spf13/cobra"
+)
+
+func NewChannelCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "channel <command>",
+		Short: "Manage channels",
+	}
+
+	fl := cmd.PersistentFlags()
+	fl.StringP("type", "t", "", "[required] Channel type such as 'messaging' or 'livestream'")
+	cmd.MarkPersistentFlagRequired("type")
+
+	cmd.AddCommand(
+		getChannelCmd(),
+		createChannelCmd(),
+		deleteChannelCmd(),
+		updateChannelCmd(),
+		listChannelsCmd())
+
+	return cmd
+}
+
+func getChannelCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "get --type [channel-type] --id [channel-id]",
+		Short: "Returns a channel",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetStreamClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chanType, _ := cmd.Flags().GetString("type")
+			chanId, _ := cmd.Flags().GetString("id")
+
+			r, err := c.Channel(chanType, chanId).Query(cmd.Context(), &stream.QueryRequest{})
+			if err != nil {
+				return err
+			}
+
+			w := tabwriter.NewWriter(cmd.OutOrStdout(), 0, 0, 2, ' ', 0)
+			t := tabby.NewCustom(w)
+			t.AddHeader("CID", "Member count", "Created By", "Created At", "Updated At", "Custom Data")
+			t.AddLine(r.Channel.CID,
+				r.Channel.MemberCount,
+				r.Channel.CreatedBy.ID,
+				r.Channel.CreatedAt.Format(time.RFC822),
+				r.Channel.UpdatedAt.Format(time.RFC822),
+				r.Channel.ExtraData)
+			t.Print()
+
+			return nil
+		},
+	}
+
+	fl := cmd.Flags()
+	fl.StringP("id", "i", "", "[required] Channel id")
+	cmd.MarkFlagRequired("id")
+
+	return cmd
+}
+
+func createChannelCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "create --type [channel-type] --id [channel-id]",
+		Short: "Creates a channel",
+		RunE: func(cmd *cobra.Command, args []string) error {
+			c, err := config.GetConfig(cmd).GetStreamClient(cmd)
+			if err != nil {
+				return err
+			}
+
+			chanType, _ := cmd.Flags().GetString("type")
+			chanId, _ := cmd.Flags().GetString("id")
+			user, _ := cmd.Flags().GetString("user")
```

Comment:
These flags are mandatory (eg.: `cmd.MarkFlagRequired("id")`), so we don't need to check for errors.